src-ui/lib/db/test_helpers.ts

```diff
@@ -0,0 +1,50 @@
+import * as schema from '$lib/drizzle/schema'
+import initSqlJs, { type Database } from 'sql.js'
+import { drizzle, SqliteRemoteDatabase } from 'drizzle-orm/sqlite-proxy'
+
+const sqlite3 = await initSqlJs()
+export function createTestDB(): {
+  drizzle: SqliteRemoteDatabase<typeof schema>
+  sqlite: Database
+} {
+  const db = new sqlite3.Database()
+
+  // NOTE: this sql schema is a hack to unblock testing in the short term. Longer term,
+  // this should be replaced by a schema.sql that is exported from the dev DB
+  // whenever a migration is completed.
+  const sqlSchema = `CREATE TABLE books (
+      id TEXT PRIMARY KEY default(uuid_blob(uuid())) NOT NULL,
+      isbn10 TEXT,
+      isbn13 TEXT,
+      title TEXT NOT NULL,
+      subtitle TEXT,
+      authors TEXT NOT NULL,
+      tags TEXT,
+      series TEXT,
+      pageCount INTEGER,
+      publicationDate TEXT,
+      copyrightDate TEXT,
+      coverImages TEXT,
+      hasRead INTEGER
+    );`
+  db.exec(sqlSchema)
+
+  const drizzleDb = drizzle(
```

Comment:
This uses an in-memory sqlite instance instead of connecting to a rust backend. This lets us write .js-only tests -- while still relying on the exact behavior of sqlite + drizzle, etc.

Minor caveat is that there could be a difference in the build-flags / extensions in sqlite in the two versions. If that happens on a feature we care about, we may need to do work to bring the two into closer alignment -- which should be possible, just takes a bit more work.